test/benchmark/cuda_driver.cpp

```diff
@@ -0,0 +1,166 @@
+/*************************************************************************
+ * Copyright (C) 2018-2022 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#ifdef NMODL_LLVM_CUDA_BACKEND
+
+#include <fstream>
+#include <regex>
+
+#include "codegen/llvm/codegen_llvm_visitor.hpp"
+#include "codegen/llvm/llvm_utils.hpp"
+#include "cuda_driver.hpp"
+#include "fmt/format.h"
+#include "utils/common_utils.hpp"
+
+#include "llvm/Bitcode/BitcodeWriter.h"
+
+using fmt::literals::operator""_format;
+
+namespace nmodl {
+namespace runner {
+
+void CUDADriver::checkCudaErrors(CUresult err) {
+    if (err != CUDA_SUCCESS) {
+        const char* ret = NULL;
+        cuGetErrorName(err, &ret);
+        throw std::runtime_error("CUDA error: " + std::string(ret));
+    }
+}
+
+void CUDADriver::checkNVVMErrors(nvvmResult err) {
+    if (err != NVVM_SUCCESS) {
+        size_t program_log_size;
+        nvvmGetProgramLogSize(prog, &program_log_size);
+        auto program_log = (char*) malloc(program_log_size);
```

Comment:
Curious- is this `malloc` way is a standard approach? what happens to allocated memory when this exception is catched somewhere?

Reply:
Improved this :+1: I think after C++17 this can become even better with the use of `str.data()`